tests/test_compiler.py

```diff
@@ -337,6 +318,7 @@ def test_generate_fields_dynamic_after_bitfield(cs: cstruct, TestEnum: Enum, oth
         Field("b", _type, 4),
         Field("c", cs.char["size"], offset=3),
     ]
+    fields[3].type.dynamic = True   # cs.char["size"] is a hack so patch dynamic
```

Comment:
What do you mean with it being a hack?

Reply:
AFAIK   `Field("c", cs.char["size"], offset=3)` does not set dynamic is True. 

Maybe mock is a better term

Reply:
Looks like faulty usage of this. `_make_array` expects an integer size or an `Expression` here (from the syntax parser). On the one hand I believe this should be allowed usage [since it's possible to add fields yourself from code](https://github.com/fox-it/dissect.cstruct/blob/e990b5301ffa6c739cdbd954baeb70d3a970a362/dissect/cstruct/types/structure.py#L354-L359). I think it can be solved by simply wrapping the string in an `Expression` in `_make_array`.

Or we don't allow this, in which case we should turn this "mock" more proper by making it `cs.char[Expression("size")]`.